assets/src/domain/eventEditor/services/filterState/tickets/useTicketsFilterBarService.ts

```diff
@@ -18,6 +19,37 @@ const useTicketsFilterBarService = (): void => {
 		registerSorter: registerTicketsSorter,
 	} = useFilterBarService<Domain, typeof ticketsList, Ticket, TFSM>(domain, ticketsList);
 
+	/**
+	 * isChained filter needs special treatment :)
+	 *
+	 * Unlike other sorters and filters, isChained filter is dependent upon
+	 * external state which can change. Thus we need to update our filter callback
+	 * to make sure the stale state is not bound to the filter callback.
+	 */
+	const [isChainedFilter, isChainedDeps] = useIsChainedFilter();
+	// To avoid multiple filter registrations, we will store the aleady registered
+	// filter unsubscribe callback in ref to use it to remove the existing filter.
+	const chainedFilterRef = useRef<VoidFunction>();
```

Comment:
interesting approach, maybe add a comment somewhere about the fact that mutating `ref` won't cause re-renders and this is used just as a global variable(kind of) which is not changed during renders.

Reply:
That's understood about and is mentioned in its [docs](https://reactjs.org/docs/hooks-reference.html#useref)